hooks/playbooks/ironic_inject_tls.yml

```diff
@@ -0,0 +1,117 @@
+---
+- name: Inject CA bundle in ironic-python-agent and create images
+  hosts: "{{ cifmw_target_hook_host | default('localhost') }}"
+  gather_facts: false
+  vars:
+    _namespace: openstack
+  environment:
+    KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
+    PATH: "{{ cifmw_path }}"
+  tasks:
+    - name: Collect facts
+      ansible.builtin.setup:
+        gather_subset:
+          - '!all'
+          - '!min'
+          - user_gid
+          - user_id
+
+    - name: Create directory
+      ansible.builtin.file:
+        path: "{{ cifmw_basedir }}/artifacts/ironic_ipa_tls/initramfs"
+        state: directory
+        mode: "0755"
+
+    - name: Get internal-ca-bundle and IPA image from k8s resources
+      ansible.builtin.set_fact:
+        internal_ca_bundle: >-
+          {{
+            query(
+              'kubernetes.core.k8s',
+              api_version='v1',
+              kind='Secret',
+              resource_name='combined-ca-bundle',
+              namespace=_namespace
+            ) | first
+              | community.general.json_query('data."internal-ca-bundle.pem"')
```

Comment:
we try to avoid `json_query` whenever possible. Would it be possible to use `select`/`selectattr` and `map` instead?

Reply:
Why do we avoid `json_query`?
I spent a bunch of time trying to figure out how to do it with select/selectattr and map. That stuff is like french to me ...

Reply:
@cjeanner - so how do we escape the `.` in the field?
```
    - name: Get internal-ca-bundle and IPA image from k8s resources
      ansible.builtin.set_fact:
        internal_ca_bundle: >-
          {{
            query(
              'kubernetes.core.k8s',
              api_version='v1',
              kind='Secret',
              resource_name='combined-ca-bundle',
              namespace=namespace
            ) | map(attribute='data') | map(attribute=['internal-ca-bundle.pem']) | first
          }}
```

this code works on other secrets to get for example `data.host` from the `rabbitmq-cell1-default-user`
```
"The task includes an option with an undefined variable. The error was: dict object has no element ['internal-ca-bundle.pem'].
```

Reply:
we can address this later :) there are already other occurrencies of json_query in the repo.